WooCommerce/Classes/Yosemite/DefaultStoresManager.swift

```diff
@@ -537,13 +570,32 @@ private extension DefaultStoresManager {
             ServiceLocator.selectedSiteSettings.refresh()
             ServiceLocator.shippingSettingsService.update(siteID: siteID)
         }
-        retrieveOrderStatus(with: siteID)
         synchronizePaymentGateways(siteID: siteID)
         synchronizeAddOnsGroups(siteID: siteID)
-        synchronizeSystemPlugins(siteID: siteID)
         synchronizeSitePlugins(siteID: siteID)
 
         sendTelemetryIfNeeded(siteID: siteID)
+
+        Task { @MainActor in
+            // Order statuses and system plugins syncing are required outside of snapshot tracking.
+            async let orderStatuses = retrieveOrderStatus(with: siteID)
+            async let systemPlugins = synchronizeSystemPlugins(siteID: siteID)
+
+            let snapshotTracker = SiteSnapshotTracker(siteID: siteID)
+            guard let orderStatuses = await orderStatuses, let systemPlugins = await systemPlugins else {
+                return
+            }
+            guard snapshotTracker.needsTracking() else {
+                return
+            }
+            // Only fetches number of products when snapshot tracking is needed.
+            guard let numberOfProducts = await retrieveNumberOfProducts(siteID: siteID) else {
+                return
+            }
+            snapshotTracker.trackIfNeeded(orderStatuses: orderStatuses,
+                                          numberOfProducts: numberOfProducts,
+                                          systemPlugins: systemPlugins)
+        }
```

Comment:
nit: I think we can wrap this into a separate function to make the `restoreSessionSiteIfPossible` method easier to read!

Reply:
Good idea, I moved the code that's related to snapshot tracking to a separate function https://github.com/woocommerce/woocommerce-ios/pull/10484/commits/eb68c6853664e33686170526a320c379f573aeb7. Hopefully this also makes it more clear when removing the tracking code!